cri/v1alpha1/cri.go

```diff
@@ -462,6 +465,11 @@ func (c *CriManager) ListPodSandbox(ctx context.Context, r *runtime.ListPodSandb
 		return nil, fmt.Errorf("failed to list sandbox: %v", err)
 	}
 
+	sandboxList, err = c.filterInvalidSandboxes(ctx, sandboxList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to filter invalid sandboxes: %v", err)
+	}
+
 	sandboxes := make([]*runtime.PodSandbox, 0, len(sandboxList))
 	for _, s := range sandboxList {
 		sandbox, err := toCriSandbox(s)
```

Comment:
How about put the outermost loop of `filterInvalidSandboxes` here, WDYT?

Expect that, LGTM.

Reply:
Hmm... It's a code organization issue, I want to encapsulate all the logic into this function.

Now, I think it's OK.

As the code evolve, maybe it will be better to put the loop outside :)